src/Features/Core/Portable/Completion/PatternMatchHelper.cs

```diff
@@ -18,11 +18,25 @@ namespace Microsoft.CodeAnalysis.Completion
     /// </summary>
     internal sealed class PatternMatchHelper(string pattern) : IDisposable
     {
+        private static readonly CultureInfo EnUSCultureInfo;
+
+        static PatternMatchHelper()
+        {
+            try
+            {
+                EnUSCultureInfo = new("en-US");
+            }
+            catch (CultureNotFoundException)
+            {
+                // See https://github.com/microsoft/vscode-dotnettools/issues/386
+                // This can happen when running on a runtime that is setup for culture invariant mode only.
+                EnUSCultureInfo = CultureInfo.InvariantCulture;
```

Comment:
Does having this replaced in break any of the other code expecting this to be en-US? For example, the check here:

```
if (culture.ThreeLetterWindowsLanguageName.Equals(EnUSCultureInfo.ThreeLetterWindowsLanguageName))
```

That doesn't do anything strange now?

Reply:
From a basic test it at least doesn't through
```
Console.WriteLine(CultureInfo.InvariantCulture.ThreeLetterWindowsLanguageName);
Console.WriteLine(CultureInfo.CurrentUICulture.ThreeLetterWindowsLanguageName.Equals(CultureInfo.InvariantCulture.ThreeLetterWindowsLanguageName));
```
writes

```
IVL
False
```

Docs also say this (though not 100% the same)
```
It can be used in almost any method in the [System.Globalization](https://learn.microsoft.com/en-us/dotnet/api/system.globalization?view=net-7.0) namespace that requires a culture. The objects returned by properties such as [CompareInfo](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.compareinfo?view=net-7.0), [DateTimeFormat](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.datetimeformat?view=net-7.0), and [NumberFormat](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.numberformat?view=net-7.0) also reflect the string comparison and formatting conventions of the invariant culture.
```

Reply:
> That doesn't do anything strange now?

It may.  But i don't believe that's somethign we should care/invest in.  This gets us out of server crashing, which is the pri0 issue.  Having slightly incorrect sorting/filtering  in pattern matching on this niche platform is pri4/sev4 if it happens.